test/unit-test/ota_utest.c

```diff
@@ -467,6 +467,31 @@ static OtaMqttStatus_t stubMqttPublish( const char * const unused_1,
     return OtaMqttSuccess;
 }
 
+static OtaMqttStatus_t stubMqttPublishOnlySuccedsTopicIsCorrect( const char * const topic,
+                                                                 uint16_t topicLength,
+                                                                 const char * unused_1,
+                                                                 uint32_t unused_2,
+                                                                 uint8_t unused_3 )
+{
+    ( void ) unused_1;
+    ( void ) unused_2;
+    ( void ) unused_3;
+
+    /* Maximum topic size is 128 characters for IoT Core */
+    TEST_ASSERT_LESS_OR_EQUAL( 128U, topicLength );
+    TEST_ASSERT_GREATER_THAN( 1U, topicLength );
+
+    char expected[ 129 ] = { 0 };
```

Comment:
Note: this could be even tighter. Left a bit relaxed incase we change the default topic name